crates/gateway-types/src/reply.rs

```diff
@@ -389,6 +399,103 @@ pub mod transaction {
         }
     }
 
+    #[derive(Copy, Clone, Default, Debug, PartialEq, Eq, Dummy)]
+    #[repr(u8)]
+    pub enum DataAvailabilityMode {
+        #[default]
+        L1 = 0,
+    }
+
+    impl Serialize for DataAvailabilityMode {
+        fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error>
+        where
+            S: serde::Serializer,
+        {
+            serializer.serialize_u8(*self as u8)
+        }
+    }
```

Comment:
```suggestion
    #[derive(Copy, Clone, Default, Debug, PartialEq, Eq, Dummy)]
    pub enum DataAvailabilityMode {
        #[default]
        L1,
    }

    impl Serialize for DataAvailabilityMode {
        fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error>
        where
            S: serde::Serializer,
        {
            match self {
                Self::L1 => serializer.serialize_u8(0),
            }
        }
    }
```

Reply:
Why are they using integers here in any case 🤔 weird decision

Reply:
It is -- but on the other hand they're using the integer representation of these values in the transaction hash computation so there has to be a canonical representation for them.